src/paragraph/Paragraph.tsx

```diff
@@ -7,59 +7,68 @@ import { StyledParagraph } from './Paragraph.style'
 
 const DEFAULT_MAX_CHAR_SIZE = 180
 
-export type ParagraphProps = Readonly<{
+type ParagraphBaseProps = Readonly<{
   className?: string
-  children: string
-  isExpandable?: boolean
-  expandLabel?: string
   itemProp?: string
 }>
 
-type ParagraphState = {
-  isExpanded: boolean
+export type ParagraphSimpleProps = ParagraphBaseProps &
+  Readonly<{
+    children: string
+  }>
+
+export type ParagraphExpandableProps = ParagraphBaseProps &
+  Readonly<{
+    isExpandable: true
+    expandLabel: string
+    children: string
+  }>
+
+export type ParagraphProps = ParagraphSimpleProps | ParagraphExpandableProps
+
+const truncateText = (text: string) => `${text.substring(0, DEFAULT_MAX_CHAR_SIZE)}…`
+
+function SimpleParagraph(props: ParagraphSimpleProps): JSX.Element {
+  const { className, itemProp, children } = props
+
+  return (
+    <StyledParagraph className={cc(className)} role="presentation">
+      <Text itemProp={itemProp} tag={TextTagType.PARAGRAPH} newlineToBr>
+        {children}
+      </Text>
+    </StyledParagraph>
+  )
 }
 
-export class Paragraph extends PureComponent<ParagraphProps> {
-  static defaultProps: Partial<ParagraphProps> = {
-    className: '',
-    isExpandable: false,
-    expandLabel: '',
-    itemProp: null,
-  }
+function ExpandableParagraph(props: ParagraphExpandableProps): JSX.Element {
+  const { className, itemProp, expandLabel, children: originalContent } = props
 
-  state: ParagraphState = {
-    isExpanded: false,
-  }
+  const [isExpanded, setIsExpanded] = useState(originalContent.length < DEFAULT_MAX_CHAR_SIZE)
 
-  expandParagraph = () => {
-    this.setState({ isExpanded: true })
+  const expandParagraph = () => {
+    setIsExpanded(true)
   }
 
-  getTruncatedText() {
-    return `${this.props.children.substring(0, DEFAULT_MAX_CHAR_SIZE)}…`
-  }
+  const content = isExpanded ? originalContent : truncateText(originalContent)
 
-  render() {
-    const { className, children: originalContent, isExpandable, expandLabel, itemProp } = this.props
-    const { isExpanded } = this.state
-
-    const isContentTruncated =
-      isExpandable && !isExpanded && originalContent.length > DEFAULT_MAX_CHAR_SIZE
-    const content = isContentTruncated ? this.getTruncatedText() : originalContent
-
-    const readMoreButton = isContentTruncated && (
-      <Button status={ButtonStatus.UNSTYLED} onClick={this.expandParagraph} className="mt-s">
-        {expandLabel}
-      </Button>
-    )
-
-    return (
-      <StyledParagraph className={cc(className)} role="presentation">
-        <Text {...(itemProp && { itemProp })} tag={TextTagType.PARAGRAPH} newlineToBr>
-          {content}
-        </Text>
-        {readMoreButton}
-      </StyledParagraph>
-    )
+  return (
+    <StyledParagraph className={cc(className)} role="presentation">
+      <Text itemProp={itemProp} tag={TextTagType.PARAGRAPH} newlineToBr>
```

Comment:
Since you are on the refactor mode, could you please consider using typographic components `TextBody` instead of `Text`. 

```
<TextBody as="p" />
```

You don't need the `newlinetoBr` attr since  `TextBody` already treat breaking lines with CSS

`white-space: pre-line;`